File: javascript/src/agent/setup/names.ts
/**
 * safeName maps a repo or file path to an ADK-agent-name-safe string (every
 * non-alphanumeric character becomes `_`). Shared by the workflow agents that derive a
 * sub-agent name from a path (fixflow analyze, summary fetchers). Mirrors the Go
 * reference's setup.SafeName.
 */
export function safeName(s: string): string {
  return [...s].map((ch) => (/[a-zA-Z0-9]/.test(ch) ? ch : '_')).join('');
}

File: kotlin/src/main/kotlin/io/github/jkjamies/automationagent/agent/setup/Names.kt
/*
 * Package setup holds shared utilities for building agents. safeName lives here so the
 * workflow agents that derive an ADK sub-agent name from a repo or file path (fixflow
 * analyze, summary fetchers) share one implementation. Mirrors the Go reference's
 * setup.SafeName.
 */
package io.github.jkjamies.automationagent.agent.setup

/** Replaces every non-ASCII-alphanumeric character with `_`, for a safe ADK agent name. */
internal fun safeName(s: String): String =
    s.map { c -> if (c in 'a'..'z' || c in 'A'..'Z' || c in '0'..'9') c else '_' }.joinToString("")
